app/src/main/java/com/example/encoratask/ItemListActivity.kt

```diff
@@ -43,23 +56,27 @@ class ItemListActivity : AppCompatActivity() {
                     .setAction("Action", null).show()
         }
 
-        setupRecyclerView(findViewById(R.id.item_list))
+
+        viewModel.listCharacter.observe(this, Observer {
+            setupRecyclerView(findViewById(R.id.item_list), it)
+        })
     }
 
-    private fun setupRecyclerView(recyclerView: RecyclerView) {
-        recyclerView.adapter = SimpleItemRecyclerViewAdapter(DummyContent.ITEMS)
+    private fun setupRecyclerView(recyclerView: RecyclerView, list: List<Character>) {
+        recyclerView.adapter = SimpleItemRecyclerViewAdapter(list)
     }
 
-    class SimpleItemRecyclerViewAdapter(private val values: List<DummyContent.DummyItem>) :
+    class SimpleItemRecyclerViewAdapter(private val values: List<Character>) :
             RecyclerView.Adapter<SimpleItemRecyclerViewAdapter.ViewHolder>() {
 
         private val onClickListener: View.OnClickListener
 
         init {
             onClickListener = View.OnClickListener { v ->
-                val item = v.tag as DummyContent.DummyItem
+                val item = v.tag as Character
                 val intent = Intent(v.context, ItemDetailActivity::class.java).apply {
-                    putExtra(ItemDetailFragment.ARG_ITEM_ID, item.id)
+                    putExtra("NAME", item.name)
```

Comment:
Two comments here:
1. Why do you remove the id to add name and gender?
2. If you're going forward to keep on name and gender, I strongly suggest you to add this strings "NAME" and "GENDER" to the companion block as a good practice